lib/http2/hpack.c

```diff
@@ -299,6 +301,11 @@ int h2o_hpack_decode_header(h2o_mem_pool_t *pool, void *_hpack_header_table, h2o
     int64_t index = 0;
     int value_is_indexed = 0, do_index = 0;
 
+    /* This function might "leak" newly allocated memory block, assuming that the memory is from the pool and will be freed
+     * soon at pool destruction.
+     * This assersion is to make sure that we always specify a pool when using this function. */
+    assert(pool != NULL);
```

Comment:
I'm not sure if I like the idea of adding assertions due to how Coverity (or any analysis tool) works. If we continue doing that, the code would become hard to maintain.

Is there an issue with marking this as a non-issue in Coverity and moving on?

Reply:
I think your concern makes sense. I'll revert this and simply mark the issue as false positive.

Reply:
Reverted.